providers/implementations/keymgmt/mlx_kmgmt.c

```diff
@@ -710,9 +764,19 @@ static void *mlx_kem_gen(void *vgctx, OSSL_CALLBACK *osslcb, void *cbarg)
     if ((gctx->selection & OSSL_KEYMGMT_SELECT_KEYPAIR) == 0)
         return key;
 
-    /* For now, using the same "propq" for all components */
-    key->mkey = EVP_PKEY_Q_keygen(key->libctx, key->propq,
-                                  key->minfo->algorithm_name);
+    adjusted_propq = get_adjusted_propq(propq);
```

Comment:
Again missing null check and error return

Reply:
And why 'propq' and not 'key->propq' here?
That means we are looking at the gctx value of propq and not the one we are currently passing into the other functions - that seems incorrect.

Reply:
'key->propq' is assigned from 'propq' in `mlx_kem_key_new`